doc/release/upcoming_changes/27735.new_feature.rst

```diff
@@ -0,0 +1,7 @@
+* UFuncs now support ``__dict__`` attribute and allow overriding ``__doc__``
+  (either directly or via ``ufunc.__dict__["__doc__"]``). ``__dict__`` can be
+  used to also override other properties, such as ``__module__`` or
+  ``__qualname__``.
+
+* ``_add_newdoc_ufunc`` is now deprecated. ``ufunc.__doc__ = newdoc`` should
+  be used instead.
```

Comment:
Nit, but it should be split into two files (different categories).

Reply:
Ah right, done!